py/embedded-server/deephaven_server/cli.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+#
+import click
+import signal
+import sys
+import webbrowser
+
+from .server import Server
+
+@click.group()
+def cli():
+    """
+    Command-line interface entry point for the Deephaven embedded server application.
+    Accepts a command to start the application and the arguments to use.
+    """
+    pass
+
+
+@cli.command()
+@click.option("--host", default=None, help="The host to bind to.")
+@click.option("--port", default=None, type=int, help="The port to bind to.")
+@click.option(
+    "--key",
+    default=None,
+    help="Start the server in pre-shared key mode. If the key is empty, a random key will be used. Cannot be used if anonymous is specified.",
```

Comment:
> Cannot be used if anonymous is specified.

Why not? Aside from limitations of the patch, this is a reasonable thing to do and supported by the server? The authorization APIs would need to be used in some way to really be helpful ("anon users can access everything except the console, psk users can access everything"), but none of these are mutually exclusive.

Reply:
Removed this option.